platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/OutputWire.java

```diff
@@ -0,0 +1,234 @@
+        }
+    }
+
+    /**
+     * Get the name of this output wire. If this object is a task scheduler, this is the same as the name of the task
+     * scheduler.
+     *
+     * @return the name
+     */
+    @NonNull
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Get the wiring model that contains this output wire.
+     *
+     * @return the wiring model
+     */
+    protected WiringModel getModel() {
+        return model;
+    }
+
+    /**
+     * Forward output data to any wires/consumers that are listening for it.
+     *
+     * @param data the output data to forward
+     */
+    protected void forward(@NonNull final O data) {
+        for (final Consumer<O> destination : forwardingDestinations) {
+            try {
+                destination.accept(data);
+            } catch (final Exception e) {
+                logger.error(
+                        EXCEPTION.getMarker(),
+                        "Exception thrown on output wire {} while forwarding data {}",
+                        name,
+                        data,
+                        e);
+            }
+        }
+    }
+
+    /**
+     * Specify an input wire where output data should be passed. This forwarding operation respects back pressure.
+     *
+     * <p>
+     * Soldering is the act of connecting two wires together, usually by melting a metal alloy between them. See
+     * <a href="https://en.wikipedia.org/wiki/Soldering">wikipedia's entry on soldering</a>.
+     *
+     * <p>
+     * Forwarding should be fully configured prior to data being inserted into the system. Adding forwarding
+     * destinations after data has been inserted into the system is not thread safe and has undefined behavior.
+     *
+     * @param inputWire the input wire to forward output data to
+     * @return this
+     */
+    @NonNull
+    public final OutputWire<O> solderTo(@NonNull final InputWire<O, ?> inputWire) {
+        model.registerEdge(name, inputWire.getTaskSchedulerName(), inputWire.getName(), false);
+        forwardingDestinations.add(inputWire::put);
+        return this;
+    }
+
+    /**
+     * Specify an input wire where output data should be forwarded via injection. Injection ignores back pressure at the
+     * destination.
+     *
+     * <p>
+     * Soldering is the act of connecting two wires together, usually by melting a metal alloy between them. See
+     * <a href="https://en.wikipedia.org/wiki/Soldering">wikipedia's entry on soldering</a>.
+     *
+     * <p>
+     * Forwarding should be fully configured prior to data being inserted into the wire. Adding forwarding destinations
+     * after data has been inserted into the wire is not thread safe and has undefined behavior.
+     *
+     * @param inputWire the inputWire to forward output data to
+     * @return this
+     */
+    @NonNull
+    public OutputWire<O> injectionSolderTo(@NonNull final InputWire<O, ?> inputWire) {
```

Comment:
maybe its better if we overload `solderTo` with another parameter for injecting

Reply:
done